utils/docker-compose/README.md

```diff
@@ -0,0 +1,227 @@
+```
+./scripts/bootstrap.sh
+```
+1. Build the peer image in `utils/docker/peer` directory which is defined by the
+peer [Dockerfile](../docker/peer/Dockerfile). This step
+assumes you have already built the [fabric-private-chaincode base image](../docker/base/Dockerfile).
+Take a look at building the docker dev environment in the main [README](../../README.md#docker).
+After you have created the base image, run the following to create a modified
+peer image and the plugins necessary to start the peer.  `$FPC_PATH` is the
+location fabric-private-chaincode repository on your host machine.
+```
+cd $FPC_PATH/utils/docker/peer
+docker build -t hyperledger/fabric-peer-fpc .
+```
+By default the image will clone the master branch on
+https://github.com/hyperledger-labs/fabric-private-chaincode. If you want to use
+a different fork of the repo or a different branch you provide
+FPC_REPO_URL and FPC_REPO_BRANCH as build args.
+```
+cd $FPC_PATH/utils/docker/peer
+docker build -t hyperledger/fabric-peer-fpc --build-arg FPC_REPO_URL=<repo-url> --build-arg FPC_REPO_BRANCH=<repo-branch> .
+```
+If you want to build the peer image using your local copy of your repo you can
+use the same build args, but specify `file:///tmp/build-src/.git` as the
+`FPC_REPO_URL`. You will also need to create the image at the root of this repo
+so that the local repo will be in the build context for the docker daemon.
+```
+cd $FPC_PATH
+docker build -t hyperledger/fabric-peer-fpc -f utils/docker/peer/Dockerfile --build-arg FPC_REPO_URL=file:///tmp/build-src/.git --build-arg FPC_REPO_BRANCH=$(git rev-parse --abbrev-ref HEAD) .
+```
+2. Generate the cryptographic material needed for the network by running the
+[generate](scripts/generate.sh) script. Cryptogen will be used to generate all the
+credentials needed based on the configuration filesabove and place them in the
+`network-config/crypto-config` directory.  Configtxgen will be used to create
+the genesis block which is used to start up the orderer as well as the peer
+create channel configuration transaction. These will be placed in the
+`network-config/config` directory. The `crypto-config` & `config` directory will
+be mounted into every container of the FPC network as specified in the
+docker-compose file. **This script is not
+idempotent and will delete the contents of `crypto-config` & `config` when run
+to ensure a clean start.**
+```
+cd $FPC_PATH/utils/docker-compose
+scripts/generate.sh
+```
+3. Start the network. Run the [start](scripts/start.sh) script. This will use
+docker-compose to start the network as well as starting the channel `mychannel`.
+By default, this script will use FPC peers. If non FPC peers are desired, set
+`$USE_FPC` to `false`.
+```
+.scripts/start.sh
+```
+
+## Deploying your FPC Chaincode
+The examples directory has been mounted into the peer container for convenience,
+under `/opt/examples`.
+
+1. Follow the [steps](../../examples/README.md) in the tutorial to build your
```

Comment:
The hello world example requires quite some manual steps, so to some extent if you are only interested in the "delta" of the docker-compose, this is a bit painful.
When i tested, i just used the echo cc which worked also fine.  Probably not worth effort to change it now, but maybe point that out in text?